tests/unit/lib/foxbox/foxbox_test.js

```diff
@@ -0,0 +1,84 @@
+import Foxbox from 'js/lib/foxbox/foxbox';
+
+describe('Foxbox >', function() {
+  let foxbox;
+  const singleBox = [{
+    public_ip: '1.1.1.1',
+    client: 'abc',
+    message: JSON.stringify({
+      local_origin: 'https://local.abc.box.knilxof.org:3000',
+      tunnel_origin: 'null',
+    }),
+    timestamp: Math.floor(Date.now() / 1000),
+  }];
+  const multiBox = singleBox.concat([
+    {
+      public_ip: '2.2.2.2',
+      client: 'def',
+      message: JSON.stringify({
+        local_origin: 'http://local.def.box.knilxof.org:3000',
+        tunnel_origin: 'null',
+      }),
+      timestamp: Math.floor(Date.now() / 1000),
+    },
+  ]);
+
+  function getMockNet(single = true) {
+    return {
+      init: sinon.stub(),
+      fetchJSON: () => Promise.resolve(single ? singleBox : multiBox),
+    };
+  }
+
+  describe('constructor >', function() {
+    it('exposes a non extensible object', function() {
+      foxbox = new Foxbox();
+
+      assert.isObject(foxbox);
+      assert.throws(() => foxbox.newProp = true);
+    });
+
+    it('a net parameter can be provided', function(done) {
+      const mockNet = getMockNet();
+      foxbox = new Foxbox({ net: mockNet });
+
+      foxbox.init()
+        .then(() => {
+          sinon.assert.calledOnce(mockNet.init);
+        })
+        .then(done, done);
+    });
+  });
+
+  describe('_initDiscovery >', function() {
```

Comment:
Yep, I agree, we actually want to test `init` here.
Regarding the test plan I also agree but that's out of the scope of this PR. I simply want to make sure that we don't regress with the code related to the registration server and the box discovery for now.